python/lsst/sims/utils/ModifiedJulianDate.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+import warnings
+import os
+
+from astropy.time import Time
+
+from lsst.utils import getPackageDir
+
+__all__ = ["ModifiedJulianDate"]
+
+class ModifiedJulianDate(object):
+
+
+    def __init__(self, TAI=None, UTC=None):
+        """
+        Must specify either:
+
+        @param [in] TAI = the International Atomic Time as an MJD
+
+        or
+
+        @param [in] UTC = Universal Coordinate Time as an MJD
+        """
+
+        if TAI is None and UTC is None:
+            raise RuntimeError("You must specify either TAI or UTC to "
+                               "instantiate ModifiedJulianDate")
+
+        if TAI is not None:
+            self._time = Time(TAI, scale='tai', format='mjd')
+            self._tai = TAI
+            self._utc = None
+        else:
+            self._time = Time(UTC, scale='utc', format='mjd')
+            self._utc = UTC
+            self._tai = None
+
+        self._tt = None
+        self._tdb = None
+        self._ut1 = None
+        self._dut1 = None
+
+
+    def __eq__(self, other):
+        return self._time == other._time
+
+
+    @property
+    def TAI(self):
+        """
+        International Atomic Time as an MJD
+        """
+        if self._tai is None:
+            intermediate_value = self._time.tai
+
+            try:
+                self._tai = intermediate_value.value
+            except:
+                self._tai = intermediate_value
+
```

Comment:
I'm not quite sure what's happening here. Why do sometimes you look for the intermediate_value.value and sometimes for the intermediate_value itself?